model/src/test/kotlin/config/IncludesTest.kt

```diff
@@ -0,0 +1,245 @@
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.CuratedPackage
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.Repository
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.config.config.PathInclude
+import org.ossreviewtoolkit.model.config.config.PathIncludeReason
+
+class IncludesTest : WordSpec() {
+    private val id = Identifier("type", "namespace", "name", "version")
+
+    private val pkg = CuratedPackage(metadata = org.ossreviewtoolkit.model.Package.EMPTY.copy(id = id))
+
+    private val projectId1 = id.copy(name = "project1")
+    private val projectId2 = id.copy(name = "project2")
+    private val project1 = Project.EMPTY.copy(id = projectId1, definitionFilePath = "path1")
+    private val project2 = Project.EMPTY.copy(id = projectId2, definitionFilePath = "path2")
+
+    private val pathInclude1 = PathInclude("path1", PathIncludeReason.SOURCE_OF, "")
+    private val pathInclude2 = PathInclude("path2", PathIncludeReason.SOURCE_OF, "")
+    private val pathInclude3 = PathInclude("path3", PathIncludeReason.SOURCE_OF, "")
+
+    private val scope1 = Scope("scope1", setOf(PackageReference(id)))
+    private val scope2 = Scope("scope2", setOf(PackageReference(id)))
+    private val scopeProject1 = Scope("scopeProject1", setOf(PackageReference(project1.id)))
+
+    private lateinit var ortResult: OrtResult
+
+    override suspend fun beforeEach(testCase: TestCase) {
+        ortResult = OrtResult(
+            repository = Repository.EMPTY,
+            analyzer = AnalyzerRun.EMPTY
+        )
+    }
+
+    private fun setIncludes(paths: List<PathInclude> = emptyList()) {
+        setIncludes(Includes(paths = paths))
+    }
+
+    private fun setIncludes(includes: Includes) {
+        val config = ortResult.repository.config.copy(includes = includes)
+        ortResult = ortResult.replaceConfig(config)
+    }
+
+    private fun setProjects(vararg projects: Project) {
+        val packages = mutableSetOf<Package>()
+        if (id in projects.flatMap { ortResult.dependencyNavigator.projectDependencies(it) }) packages += pkg.metadata
+        val analyzerResult = ortResult.analyzer!!.result.copy(
+            projects = projects.toSet(),
+            packages = packages
+        )
+        ortResult = ortResult.copy(analyzer = ortResult.analyzer!!.copy(result = analyzerResult))
+    }
+
+    init {
+        "isExcluded" should {
+            "return false if the package is not included but no includes are defined" {
+                setProjects(
+                    project1.copy(scopeDependencies = setOf(scope1))
+                )
+
+                ortResult.isExcluded(id) shouldBe false
```

Comment:
Renaming `id` to `packageId` would make it easier to understand these test cases, I was confused first because I though it would refer to a project.